src/ripple/app/tx/impl/SetTrust.cpp

```diff
@@ -183,18 +191,19 @@ SetTrust::doApply()
 
     auto viewJ = ctx_.app.journal("View");
 
-    if (account_ == uDstAccountID)
+    // Trust lines to self are impossible but because of the old bug there are
+    // two on 19-02-2022. This code was here to allow those trust lines to be
+    // deleted. The fixTrustLinesToSelf fix amendment will remove them when it
+    // enables so this code will no longer be needed.
+    if (!view().rules().enabled(fixTrustLinesToSelf) &&
```

Comment:
I would drop `!view().rules().enabled(fixTrustLinesToSelf)` because I don't know if all trust lines to self have been deleted. I feel that this is less defensive as it could be.

Reply:
- Can't drop that because there are currently two trustlines to self. Without the check, servers with and without this PR would behave differently.

Reply:
The code says that if `fixTrustLinesToSelf` is disabled and `account_ == uDstAccountID`, we can delete a self-issued trustline. Once `fixTrustLinesToSelf` is enabled, no other self-issued trustlines will be deleted. This is fine as long as we know that there are **only** two self-issued trustlines on the ledger.